src/silx/gui/widgets/FrameBrowser.py

```diff
@@ -360,53 +399,37 @@ def value(self):
         """Get selected value"""
         return self._slider.value()
     
-    def spinBoxFrameRate(self) -> qt.QSpinBox:
-        """Returns the SpinBox widget for FrameRate display."""
-        return self._menuPlaySlider._framerate_action._spinbox
-
-    def spinBoxInterval(self) -> qt.QSpinBox:
-        """Returns the SpinBox widget for interval display."""
-        return self._menuPlaySlider._interval_action._spinbox
-
     def setFrameRate(self, value: int):
-        """Set the FrameRate value for the PlaySlider"""
-        self.spinBoxFrameRate().setValue(value)
+        """Set the frame rate at which images are displayed"""
+        self._menuPlaySlider.setFrameRate(value)
         
     def getFrameRate(self) -> int:
-        """Returns the value from the FrameRate SpinBox."""
-        return int(self.spinBoxFrameRate().value())
+        """Get the frame rate at which images are displayed"""
+        return self._menuPlaySlider.getFrameRate()
     
-    def setInterval(self, value: int):
-        """Set the Interval value for the PlaySlider"""
-        self.spinBoxInterval().setValue(value)
+    def setPlayImageStep(self, value: int):
+        """Set the step between displayed images when playing"""
+        self._menuPlaySlider.setStep(value)
     
-    def getInterval(self) -> int:
-        """Returns the value from the Interval SpinBox."""
-        return int(self.spinBoxInterval().value())
+    def getPlayImageStep(self) -> int:
+        """Returns the step between displayed images"""
+        return self._menuPlaySlider.getStep()
+
+    def _frameRateChanged(self, framerate: int):
+        """Update the timer interval"""
+        self.__timer.setInterval(int(1 / framerate * 1e3))
             
-    def _playStopSequence(self):
+    def _playButtonToggled(self, checked: bool):
         """Start/Stop the slider sequence."""
-        if self.__timer.isActive():
-            self._stopTimer()
-        else:
-            self._startTimer()
-        
+        if checked:
+            self.__timer.start()
+            return
+        self.__timer.stop()
```

Comment:
Rework the QTimer and button to make it simpler: use the check state of the button to handle the QTimer